packages/fxa-event-broker/README.md

```diff
@@ -1,5 +1,81 @@
 # fxa-event-broker
 
+The Firefox Accounts Event Broker (fxa-event-broker) is a webhook event delivery
+system for events in Firefox Accounts. These events are currently consumed from an
+internal priviledged queue that FxA services publish to, screened to determine what
+Relying Parties (RPs) should receive them, then distributed via webhooks using the
+[Security Event Token (SET)][set] format.
+
+This event broker also stores event metadata, tracking which RPs a user has logged
+into to screen event delivery to relevant RPs.
+
+## Event Format
+
+A relying party will get webhook calls for events. These events are encoded in
+SETs with the following formats. See the [SET RFC][set] for definitions and other
+examples.
+
+### Subscription State Change
+
+Sent when a users subscription state has changed to RPs that provide the changed
+subscription capability.
+
+**NOTE**: There are strict requirements about subscription state change handling
+based on the `changeTime` as documented below.
+
+- Event Identifier
+  - `https://schemas.accounts.firefox.com/event/subscription-state-change`
+- Event Payload
+  - [Subscription Event Identifier]
+    - capabilities
+      - List of subscription capabilities
+    - isActive
+      - Boolean indicating if the subscription should be considered active or not
+        for the subscription capabilities provided.
+    - changeTime
+      - Time in milliseconds when the state change occured in the payment system.
+        This value MUST be tracked by the receiving system, and events with a
+        changeTime older than the last tracked time MUST be discarded.
```

Comment:
If messages can arrive out of order, should we future-proof this event system by making a timestamp mandatory on all event types? (Fine to defer to a follow-up issue)

Reply:
There is a timestamp, but the timestamp is the generation of the security event token, which does not correspond to the originating event generation this reflects way back in Stripe. The Stripe event time is what appears here as `changeTime` which is the value that needs to be used to determine if a message should be ignored. Since these message may be not only out of order, but also encounter delays it didn't make sense to use that value as the SET's `iat`, as that might be confusing to debug and seeing tokens you were just sent with times 20+ seconds in the past. Thus adding the value here separately purely for this use-case.